app/components/post-new-form.js

```diff
@@ -1,18 +1,50 @@
 import Ember from 'ember';
 
+/**
+  The post-new-form component is used for creating new posts. It includes the
+  post type, title, editor and preview
+
+  ## default usage
+
+  ```handlebars
+  {{post-new-form post=newPost savePost='savePostMethod'}}
+  ```
+
+  @class post-new-form
+  @module Component
+  @extends Ember.Component
+ */
 export default Ember.Component.extend({
   classNames: ['post-new-form'],
   classNameBindings: ['post.postType'],
   tagName: 'form',
 
+  /**
+    @property credentials
+    @type Ember.Service
+   */
   credentials: Ember.inject.service(),
 
+  /**
+    Holds the different plcaeholder messages for each type of post
+    (task/issue/idea).
```

Comment:
`placeholder` is misspelled. Maybe reword to "for each post type: task, issue, idea, etc."